### ISSUE TITLE
Floating window will not be deleted when no listed buffers exist

### ISSUE THREAD

**Issue by bryall** [neovim/neovim#12895]
NVIM v0.5.0-948e625e1
macosx 10.15.6
kitty terminal

### Steps to reproduce using `nvim -u NORC`

launch neovim with no listed buffers
create a floating window with buffer that is not listed
call "bwipeout <floating buffer id>" on the buffer with the floating window

### Actual behaviour
the floating window/buffer would remain visible

### Expected behaviour
the floating window/buffer should close

**Comment by bfredl** [comment]
>  create a floating window with buffer that is not listed

Agree, but won't we want even the same with a _listed_ buffer? Or you mean that is already working as expected.

**Comment by bryall** [comment]
good point any floating window should be able to be deleted with any buffer on the main window. is there a way to tell that the buffer is in a floating window?

**Comment by bryall** [comment]
change the check to see if the buffer is contained in a floating window

**Comment by zeertzjq** [comment]
It is actually not very clear what the intended behavior should be here. Related: <https://github.com/vim/vim/issues/2029>.

**Comment by dundargoc** [comment]
If we flip the question, why shouldn't the window be closed?

**Comment by zeertzjq** [comment]
To match the behavior of deleting the buffer of a non-floating window while there are no other listed buffers.

**Comment by dundargoc** [comment]
Doh, I forgot that this was about floating window and that the issue for vim was for regular windows. My bad.

**Comment by dundargoc** [comment]
Hmmm, I guess it makes sense to follow the behavior of non-floating windows for consistency, but I'm not a fan of the behavior itself.

**Comment by justinmk** [comment]
Sounds related to https://github.com/neovim/neovim/pull/37861 , https://github.com/vim/vim/issues/19389 , which was resolved by documenting:

https://github.com/neovim/neovim/blob/6bb6b479332727852052baa6c1ea0801e776895e/runtime/doc/windows.txt#L1190-L1197
